models/user/user.go

```diff
@@ -959,7 +973,7 @@ func GetUserByName(ctx context.Context, name string) (*User, error) {
 	if len(name) == 0 {
 		return nil, ErrUserNotExist{0, name, 0}
 	}
-	u := &User{LowerName: strings.ToLower(name)}
+	u := &User{LowerName: strings.ToLower(name), Type: UserTypeIndividual}
```

Comment:
This change is wrong, 

* Before `GetUserByName` can be used to get a org owner.
* After: only `Individual` can be found.

A real example is shown below, and I believe there are far more bugs.

And that's why people keep  adding more and more buggy code like `GetUserOrOrgByName` in "Fix bug when creating pull request (#36166)"


@techknowlogick it needs a complete fix, or please revert if it's unable to fix.



<img width="1848" height="990" alt="image" src="https://github.com/user-attachments/assets/cd11ccdf-8a7f-4b6a-8695-7a3ff5dbf2eb" />